lib/Service/AccountService.php

```diff
@@ -78,6 +78,46 @@ public function findById(int $id): Account {
 		return new Account($this->mapper->findById($id));
 	}
 
+	/**
+	 * Finds a mail account by user id and mail address
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param string $userId			system user id
+	 * @param string $address			mail address (e.g. test@example.com)
+	 *
+	 * @return Account[]
+	 *
+	 * @throws ClientException
+	 */
+	public function findByUserIdAndAddress(string $userId, string $address): array {
+		// evaluate if cached accounts collection already exists
+		if (isset($this->accounts[$userId])) {
```

Comment:
$this->accounts is only populated if for the same request findByUserId was called. I'm not seeing that with the current code in server and mail.